### PR TITLE
Renaming functions in Fiber.scala

### LINKED ISSUE #1500: Consider renaming methods in Fiber
The current naming is descriptive, but a bit longwinded and maybe scary, I propose to replace

```scala
trait Fiber[F[_], E, A] {
   ....
  def join: F[Outcome[F, E, A]]
  def joinAndEmbed(onCancel: F[A])(implicit F: MonadCancel[F, E]): F[A] 
  def joinAndEmbedNever(implicit F: GenSpawn[F, E]): F[A]
}
```

with

```scala
trait Fiber[F[_], E, A] {
   ....
  def join: F[Outcome[F, E, A]]
  def joinWith(onCancel: F[A])(implicit F: MonadCancel[F, E]): F[A] 
  def joinWithNever(implicit F: GenSpawn[F, E]): F[A]
}
```

The work would involve renaming those methods, and changing the relevant tests, docs, and scaladoc. A pretty easy first contribution for someone :) (assuming the naming gets agreement)

### PR DESCRIPTION
**Pull request by jyoo980**
Should resolve #1500.

### CODE REVIEW

File: kernel/shared/src/main/scala/cats/effect/kernel/Fiber.scala [hunk 1]
```diff
@@ -60,7 +60,7 @@ trait Fiber[F[_], E, A] {
    * returned. If the fiber completes with [[Errored]], the error is raised.
    * If the fiber completes with [[Cancelled]], `onCancel` is run.
    */
-  def joinAndEmbed(onCancel: F[A])(implicit F: MonadCancel[F, E]): F[A] =
+  def joinWith(onCancel: F[A])(implicit F: MonadCancel[F, E]): F[A] =
```
Comment (jyoo980) [inline kernel/shared/src/main/scala/cats/effect/kernel/Fiber.scala:63]: Any reason why this is public? It doesn't seem to be used anywhere else but as a delegated call from `joinWithNever`

Comment (vasilmkd) [reply kernel/shared/src/main/scala/cats/effect/kernel/Fiber.scala:63]: It looks like we do not have a use for it in the codebase, but someone else just might. 😄

**Review by vasilmkd — APPROVED** [review]
Congratulations on your first contribution!

**Comment by djspiewak** [comment]
Thank you!

**Pull request merged**